plugins/flatpak/gs-flatpak-transaction.c

```diff
@@ -265,6 +302,25 @@ _transaction_operation_done (FlatpakTransaction *transaction,
 	}
 	switch (flatpak_transaction_operation_get_operation_type (operation)) {
 	case FLATPAK_TRANSACTION_OPERATION_INSTALL:
+		/* Handle special snowflake where "should-download" related refs for an installed ref
+		 * goes missing. In that case, libflatpak marks the main app ref as updatable
+		 * and then FlatpakTransaction resolves one of it's ops to install the related ref(s).
+		 *
+		 * We can depend on libflatpak till here. Since, libflatpak returns the main app
+		 * ref as updatable (instead of the related ref), we need to sync the main app's
+		 * statefor UI/UX.
+		 *
+		 * Map the current op's ref (which is related ref) to it's main app ref(which is currently
```

Comment:
nit: "which is *a* related ref", also pending space before "(which is currently".